projects/dashboards/data-atlas/src/app/accordion-vector-entry/accordion-vector-entry.component.ts

```diff
@@ -0,0 +1,148 @@
+import {Component, OnInit, ChangeDetectionStrategy, Input, ChangeDetectorRef} from '@angular/core';
+import {mergeMap, BehaviorSubject, combineLatest, of} from 'rxjs';
+import {
+    ClusteredPointSymbology,
+    colorToDict,
+    LayerCollectionLayerDict,
+    LayerCollectionService,
+    LayerDict,
+    LineSymbology,
+    PointSymbology,
+    PolygonSymbology,
+    ProjectService,
+    ProviderLayerCollectionIdDict,
+    RandomColorService,
+    Time,
+    UUID,
+    VectorLayer,
+    VectorResultDescriptorDict,
+    VectorSymbology,
+    VectorSymbologyDict,
+} from '@geoengine/core';
+import {DataSelectionService} from '../data-selection.service';
+import moment from 'moment';
+
+@Component({
+    selector: 'geoengine-accordion-vector-entry',
+    templateUrl: './accordion-vector-entry.component.html',
+    styleUrls: ['./accordion-vector-entry.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AccordionVectorEntryComponent implements OnInit {
+    @Input() collection!: ProviderLayerCollectionIdDict;
+    @Input() icon = 'class';
+
+    readonly layers$ = new BehaviorSubject<Array<LayerCollectionLayerDict>>([]);
+
+    constructor(
+        private readonly layerCollectionService: LayerCollectionService,
+        readonly projectService: ProjectService,
+        readonly dataSelectionService: DataSelectionService,
+        private readonly changeDetectorRef: ChangeDetectorRef,
+        private readonly randomColorService: RandomColorService,
+    ) {}
+
+    ngOnInit(): void {
+        this.layerCollectionService.getLayerCollectionItems(this.collection.providerId, this.collection.collectionId).subscribe((c) => {
+            const layers = [];
+            for (const item of c.items) {
+                if (item.type !== 'layer') {
+                    continue;
+                }
+
+                const layer = item as LayerCollectionLayerDict;
+                layers.push(layer);
+            }
+
+            this.layers$.next(layers);
+        });
+    }
+
+    loadData(layerListing: LayerCollectionLayerDict): void {
+        const id = layerListing.id;
+
+        this.layerCollectionService
+            .getLayer(id.providerId, id.layerId)
+            .pipe(
+                mergeMap((layer) => combineLatest([of(layer), this.projectService.registerWorkflow(layer.workflow)])),
```

Comment:
you could use `registerWorkflowForLayer`from backend or `registerAndGetLayerWorkflowId` from the layer-collection service. But if you need the layer i guess it does not matter.

Reply:
I need the layer for some metadata and also wrt. workflow metadata, I could not get rid of any step.

Reply:
Now I got it:

[8c6e7d6](https://github.com/geo-engine/geoengine-ui/pull/611/commits/8c6e7d6ae9dfcbe968e4183a75a09a577784bf49)